### DISCUSSION TITLE
Picking Fields from MQTT Pubs

### DISCUSSION THREAD

**Discussion by PeteCal** [zwave-js/zwave-js-ui#2528] [Q&A]
Something tells me the answer to this question is very simple in javascript but I know nothing about javascript so, I'm hoping for a python or some other solution.
I get a message like the one below from my thermostat.  All I want out of that message is the "id" field value (5-68-0) and the numeber after the 13th time "value" appears in the message (0). 
Is there an easy way or do I just do a lot of python string processing?
Here is the message:

```
{"id":"5-66-0-state","nodeId":5,"commandClass":66,"commandClassName":"Thermostat Operating State","endpoint":0,"property":"state","propertyName":"state","type":"number","readable":true,"writeable":false,"label":"Operating state","stateless":false,"commandClassVersion":2,"min":0,"max":255,"list":true,"states":[{"text":"Idle","value":0},{"text":"Heating","value":1},{"text":"Cooling","value":2},{"text":"Fan Only","value":3},{"text":"Pending Heat","value":4},{"text":"Pending Cool","value":5},{"text":"Vent/Economizer","value":6},{"text":"Aux Heating","value":7},{"text":"2nd Stage Heating","value":8},{"text":"2nd Stage Cooling","value":9},{"text":"2nd Stage Aux Heat","value":10},{"text":"3rd Stage Aux Heat","value":11}],"value":0,"lastUpdate":1657675116784,"nodeName":"ThermostatUS","nodeLocation":"SecondFloor"}
```

**Comment by robertsLando** [comment]
just parse it as a json?

**Reply by PeteCal** [reply]
I was afraid that was the answer.  I know nothing about json.  And I have a feeling I can't be brought up to speed in "five easy steps".
I have already done python code for 4 of the 8 ids for this thermostat.
But I have 5 more devices to go.
Maybe something will come along.  Otherwise I will continue with the python method.

**Reply by robertsLando** [reply]
> I know nothing about json. And I have a feeling I can't be brought up to speed in "five easy steps".

JSON is really easy, it's just a way to store data. There are libraryes for every programming languages to parse/serialize object to json and vice-versa. Here is an example for python: https://www.w3schools.com/python/python_json.asp

**Reply by PeteCal** [reply]
Cool.  I have finished my method of parsing using python but I have more devices so I will try your method.
